velox/vector/arrow/Bridge.cpp

```diff
@@ -511,6 +529,54 @@ VectorPtr createFlatVector(
 using WrapInBufferViewFunc =
     std::function<BufferPtr(const void* buffer, size_t length)>;
 
+VectorPtr createStringFlatVectorFromUtf8View(
+    memory::MemoryPool* pool,
+    const TypePtr& type,
+    BufferPtr nulls,
+    const ArrowArray& arrowArray,
+    WrapInBufferViewFunc wrapInBufferView) {
+  int num_buffers = arrowArray.n_buffers;
+
+  // The last C data buffer stores buffer sizes
+  auto* bufferSizes = (uint64_t*)arrowArray.buffers[num_buffers - 1];
+  std::vector<BufferPtr> stringViewBuffers(num_buffers - 3);
+
+  for (int32_t buffer_id = 2; buffer_id < num_buffers - 1; ++buffer_id) {
+    stringViewBuffers[buffer_id - 2] = wrapInBufferView(
+        arrowArray.buffers[buffer_id], bufferSizes[buffer_id - 2]);
+  }
+
+  BufferPtr stringViews =
+      AlignedBuffer::allocate<StringView>(arrowArray.length, pool);
+  auto* rawStringViews = stringViews->asMutable<uint64_t>();
```

Comment:
Would it not be easier if you got a string view pointer here already?

```
auto* rawStringViews = stringViews->asMutable<StringView>();
```

Reply:
Casting to a StringView is a much more expensive operation as compared to a uint64_t. We do not necessarily need the Velox StringView interface to import arrow buffers to the FlatVector.

Reply:
The StringView constructor scans through the data buffer to compute the size. This is more expensive than comparing directly with int32 length available in the first 4 bytes of the 128-bit Arrow utf8view (just like Velox StringView). This approach does not need to read/scan the data buffers and thus makes it more performant.

Reply:
I see. IIUC, the problem is that if the string is inline, then the StringView constructor will try to copy the prefix from the data buffer into the inline part (and hence access the data buffer). For non-inlined string, it should be equivalent:

https://github.com/facebookincubator/velox/blob/main/velox/type/StringView.h#L72

is this what you are referring to?